nixos/modules/services/misc/paperless.nix

```diff
@@ -325,6 +325,19 @@ in
       };
     };
 
+    configureNginx = lib.mkOption {
+      type = lib.types.bool;
+      default = false;
+      example = true;
+      description = "Whether to configure nginx as a reverse proxy.";
+    };
+
+    domain = lib.mkOption {
+      type = lib.types.str;
+      example = "paperless.example.com";
+      description = "Domain under which paperless will be available.";
+    };
+
```

Comment:
\* Only when the nginx option is enabled, please mention that.

Nit: I'd nest this option under `paperless.nginx.{enable,domain}`.

Also, if we're going to have this, shouldn't it set `PAPERLESS_URL` too?

Reply:
It sets already PAPERLESS_URL and I now made it unconditional from nginx.